homeassistant/components/tami4/button.py

```diff
@@ -0,0 +1,44 @@
+"""Button entities for Tami4Edge."""
+import logging
+
+from Tami4EdgeAPI import Tami4EdgeAPI
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import API, DOMAIN
+from .entity import Tami4EdgeBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+ENTITY_DESCRIPTION = ButtonEntityDescription(
+    key="boil_water",
+    translation_key="boil_water",
+    icon="mdi:kettle-steam",
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Perform the setup for Tami4Edge."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    api: Tami4EdgeAPI = data[API]
+
+    entity = Tami4EdgeBoilButton(api)
+
+    async_add_entities([entity])
+
+
+class Tami4EdgeBoilButton(Tami4EdgeBaseEntity, ButtonEntity):
```

Comment:
Will you be adding more buttons? Currently this platform is using 2 ways of declaring entities, I would prefer that we choose one of them to increase maintainability.

Reply:
What do you mean by "2 ways of declaring entities"?

Reply:
You're using both entity descriptions as just creating a specific class for an entity. I would say use one or the other.

Reply:
Ok, I've changed ButtonEntityDescription to EntityDescription

Reply:
That wasn't the change I mean.

Usually we use entity descriptions to quickly create entities. For this we create a generic entity that holds the description.

When you have less entities, or very specific ones, you create a class per entity.

Currently, you have it 50/50. You have a class specific for the boil water button, but you also use a entity description for the rest of the button. in this case I would prefer to choose one or the other. Since you won't be adding more, I would suggest moving the entity description fields as shorthand attributes in the class itself

Reply:
I'm inhering from Tami4EdgeBaseEntity which accepts an EntityDescription parameter and uses it to set other attributes.
Tami4EdgeBaseEntity is being used in other entities.
Should I change Tami4EdgeBaseEntity to work without EntityDescription as well, or would something like this be okay?

```py
class Tami4EdgeBoilButton(Tami4EdgeBaseEntity, ButtonEntity):
    """Boil button entity for Tami4Edge."""

    def __init__(self, api: Tami4EdgeAPI) -> None:
        """Initialize the button entity."""
        Tami4EdgeBaseEntity.__init__(
            self,
            api,
            EntityDescription(
                key="boil_water",
                translation_key="boil_water",
                icon="mdi:kettle-steam",
            ),
        )

    def press(self) -> None:
        """Handle the button press."""
        self._api.boil_water()
```

Reply:
If you change the super init to `super().__init__(...)` I think we can make it work

Reply:
It still needs an EntityDescription